File: Python/singleinheritance.py
class student:
    def __init__(self,name,age):
        self.name=name
        self.age=age
    def output(self):
        return(self.name,self.age)
class marks1(student):
    def __init__(self,name,age,marks):
        super().__init__(name,age)
        self.marks=marks
    def minfo(self):
        return(self.output(),"Your marks are ",self.marks)
name=str(input("Your name is "))
age=int(input("Your age is "))
marks=float(input("Your marks in percentage are "))
m=marks1(name,age,marks)
print(m.minfo())

File: CPP-1/singleinheritance.cpp
#include<iostream>
using std::cout;
using std::cin;
class student
{
    private:
    int rollno;
    char a[20];
    public:
    void rd();
    void dp();
};
class marks:public student
{
    private:
    float phy, chem, bio, math,sum,tm;
    public:
    void rd();
    void dp();
};
void student::rd()
{
    cout<<"Enter your roll no = ";
    cin>>rollno;
    cout<<"Enter you name = ";
    cin>>a;
}
void student::dp()
{
    cout<<"\nYour roll no is "<<rollno;
    cout<<"\nYour name is "<<a;
}
void marks::rd()
{
    student::rd();
    cout<<"Enter marks obtained in Physics = ";
    cin>>phy;
    cout<<"Enter marks obtained in Chemistry = ";
    cin>>chem;
    cout<<"Enter marks obtained in Biology = ";
    cin>>bio;
    cout<<"Enter marks obtained in Maths = ";
    cin>>math;
    cout<<"Enter grand total of marks in all subjects = ";
    cin>>tm;
    sum=((phy+chem+math+bio)*100)/tm;
}
void marks::dp()
{
    student::dp();
    cout<<"Marks obtained in Physics, Chemistry, Biology and Math are  =\n"<<phy<<"\n"<<chem<<"\n"<<bio<<"\n"<<math<<"\n";
    cout<<"Percentage is "<<sum;
}
int main()
{
    marks m;
    cout<<"Enter the following information\n";
    m.rd();
    m.dp();
}